R/qra_fit_convenience.R

```diff
@@ -1461,31 +1530,72 @@ get_imputed_ensemble_fits_and_predictions <- function(
       quantile_groups = quantile_groups,
       noncross = noncross,
       backend = backend)
+  }
+  
+  orig_qra_fit <- qra_fit
+
+  # do weight transfer among models
+  # save original weights for retrospective exploration
 
-    # do weight transfer among models
-    # save original weights for retrospective exploration
-    orig_qra_fit <- qra_fit
+  # leaving logic for "ew" as I found it but not understanding it... AG
+  if (combine_method != 'ew') {
     if(nrow(qra_fit$coefficients) == nrow(weight_transfer)) {
       # single weight per model
-      qra_fit$coefficients$beta <-
-        weight_transfer %*% matrix(qra_fit$coefficients$beta)
+      betas <- weight_transfer %>% 
+      dplyr::mutate(
+        betas = purrr::map(
+          weight_transfer,
+          ~ . %*% matrix(qra_fit$coefficients$beta)
+        )) %>% 
+      dplyr::select(!!!syms(weight_transfer_group_factors), betas)
     } else {
       # weight per quantile; adjust by iterating through quantile levels
       qs <- qra_fit$coefficients[[attr(qfm_train, 'quantile_name_col')]]
-      for (q in unique(qs)) {
-        row_inds <- which(qs == q)
-        qra_fit$coefficients$beta[row_inds] <-
-          weight_transfer %*% matrix(qra_fit$coefficients$beta[row_inds])
-      }
+      betas <- weight_transfer %>% 
+      dplyr::mutate(
+        betas = purrr::map(
+          weight_transfer,
+          function(wt) {
+            betas <- matrix(qra_fit$coefficients$beta)
+            for (q in unique(qs)) {
+              betas[which(qs == q)] <- wt %*% betas[which(qs == q)]
+            }
+            return(betas)
+          }
+        )) %>% 
+      dplyr::select(!!!syms(weight_transfer_group_factors), betas)      
     }
   }
 
   # obtain predictions
-  qra_forecast <- predict(
-    qra_fit,
-    imputed_qfm_test,
-    sort_quantiles = (noncross == "sort")) %>%
-      as.data.frame()
+  if (combine_method == 'ew' || !weight_transfer_per_group) {
+    qra_forecast <- predict(
+      qra_fit,
+      imputed_qfm_test,
+      sort_quantiles = (noncross == "sort")) %>%
+    as.data.frame()
+  } else {
+    qra_forecast <- weight_transfer_test %>% 
```

Comment:
Maybe it would - I guess that takes us back toward the class definition solution...

I was hoping this could get folded into a test for the whole function.

Reply:
I'm up for a simpler answer.  Maybe we can sort out what a test for the whole function would look like.